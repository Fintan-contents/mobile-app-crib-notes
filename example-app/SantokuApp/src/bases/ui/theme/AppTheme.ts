export type AppTheme = {
  /**
   * Theme name.
   */
  name: string;
  /**
   * Color settings group for this theme.
   */
  colors: {
    /**
     * Standard background color of the screen.
     */
    background: string;
    /**
     * Header background color.
     */
    backgroundHeader: string;
    /**
     * Background color of bottom tab.
     */
    backgroundBottomTab: string;
    /**
     * Background color of the snack bar used for notifications.
     */
    backgroundSnackbar: string;
    /**
     * Standard font color.
     */
    text: string;
    /**
     * Text color to be used when overlaid on a background with a different shade from the standard background color,
     * or when you want to invert the color within a selection.
     */
    textInverted: string;
    /**
     * Text color for more emphasis than standard (e.g. title)
     */
    textStrong: string;
    /**
     * Text color if you want it to be more subdued than standard (e.g. placeholders)
     */
    textWeak: string;
    /**
     * Standard border color.
     */
    border: string;
    /**
     * Border color for more emphasis than standard (e.g., underline in text entry field)
     */
    borderStrong: string;
    /**
     * Colors used for elements that you want to emphasize, such as normal buttons and selected elements.
     */
    primary: string;
    /**
     * Primary color for overlaid on a background with a different shade from the standard background color,
     * also used for expressing shading in combination with primary.
     */
    primaryVariant: string;
    /**
     * Colors used for auxiliary elements.
     */
    secondary: string;
    /**
     * Secondary color for overlay on a background with a different shade from the standard background color.
     * It is also used to express shading by placing it with secondary at the same time.
     */
    secondaryVariant: string;
    /**
     * The color to use for elements that are disabled or unselected.
     */
    inactive: string;
    /**
     * Colors used for destructive operation buttons and warnings against destructive operations (e.g. delete button)
     */
    danger: string;
    /**
     * Colors used for error messages.
     */
    error: string;
    /**
     * Colors used for warning messages.
     */
    warning: string;
  };
};
