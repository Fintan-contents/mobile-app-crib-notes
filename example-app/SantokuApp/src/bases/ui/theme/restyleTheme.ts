import {createTheme} from '@shopify/restyle';

const palette = {
  orange1: '#FF9800',
  orange2: '#FFF5E4',
  blue: '#29B6F6',
  black: '#212121',
  black2: '#000000',
  grey1: '#9E9E9E',
  grey2: '#E0E0E0',
  grey3: '#F5F5F5',
  grey4: '#CCCCCC',
  white: '#FFFFFF',
  divider: '#CBCBCB',
  textRed: '#FF4857',
};

export const restyleTheme = createTheme({
  colors: {
    orange1: palette.orange1,
    orange2: palette.orange2,
    blue: palette.blue,
    black: palette.black,
    black2: palette.black2,
    grey1: palette.grey1,
    grey2: palette.grey2,
    grey3: palette.grey3,
    white: palette.white,
    divider: palette.divider,
    textRed: palette.textRed,
    placeholder: palette.grey4,
    errorText: palette.textRed,
    activityIndicator: palette.grey2,
  },
  spacing: {
    s: 8,
    p2: 2,
    p4: 4,
    p8: 8,
    p12: 12,
    p16: 16,
    p24: 24,
    p32: 32,
    p48: 48,
    p64: 64,
  },
  breakpoints: {
    // phone: 0,
    // tablet: 768,
  },
  sizeVariants: {
    defaults: {
      width: 8,
      height: 8,
    },
    p18: {
      width: 18,
      height: 18,
    },
    p24: {
      width: 24,
      height: 24,
    },
    p40: {
      width: 40,
      height: 40,
    },
    p48: {
      width: 48,
      height: 48,
    },
    p120: {
      width: 120,
      height: 120,
    },
  },
  zIndices: {},
  borderRadii: {
    s: 8,
    p2: 2,
    p4: 4,
    p8: 8,
    p16: 16,
    p20: 20,
    p24: 24,
    p28: 28,
    p32: 32,
    p40: 40,
    p56: 56,
  },
  textVariants: {
    defaults: {
      color: 'black',
    },
    font20Bold: {
      fontSize: 20,
      fontWeight: 'bold',
    },
    font22Bold: {
      fontSize: 22,
      fontWeight: 'bold',
    },
    font18SemiBold: {
      fontSize: 18,
      fontWeight: '600',
    },
    font16Regular: {
      fontSize: 16,
    },
    font14Regular: {
      fontSize: 14,
    },
    font14SemiBold: {
      fontSize: 14,
      fontWeight: '600',
    },
    font14Bold: {
      fontSize: 14,
      fontWeight: 'bold',
    },
    font13Regular: {
      fontSize: 13,
      fontWeight: '400',
    },
    font13Bold: {
      fontSize: 13,
      fontWeight: '600',
    },
    button: {
      fontSize: 14,
      fontWeight: '600',
      color: 'white',
    },
    inputErrorText: {
      fontSize: 14,
      color: 'errorText',
    },
    h1: {
      // fontFamily: 'ShopifySans-Bold',
      // fontSize: 16,
      // lineHeight: 24,
      // color: 'black',
    },
    h2: {
      // fontFamily: 'ShopifySans-Bold',
      // fontSize: 16,
      // lineHeight: 24,
      // color: 'black',
    },
    h3: {
      // fontFamily: 'ShopifySans-Bold',
      // fontSize: 16,
      // lineHeight: 24,
      // color: 'black',
    },
    h4: {
      // fontFamily: 'ShopifySans-Bold',
      // fontSize: 16,
      // lineHeight: 24,
      // color: 'black',
    },
    h5: {
      // fontFamily: 'ShopifySans-Bold',
      // fontSize: 16,
      // lineHeight: 24,
      // color: 'black',
    },
    h6: {
      // fontFamily: 'ShopifySans-Bold',
      // fontSize: 16,
      // lineHeight: 24,
      // color: 'black',
    },
  },
  buttonVariants: {
    defaults: {
      backgroundColor: 'blue',
      borderRadius: 'p16',
      px: 'p12',
      py: 'p8',
    },
    tag: {},
  },
});

export type RestyleTheme = typeof restyleTheme;
