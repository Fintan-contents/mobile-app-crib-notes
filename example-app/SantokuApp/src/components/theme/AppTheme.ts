export type AppTheme = {
  name: string;
  colors: {
    background: string;
    backgroundHeader: string;
    backgroundBottomTab: string;
    backgroundSnackbar: string;
    text: string;
    textInverted: string;
    textStrong: string;
    textWeak: string;
    border: string;
    borderStrong: string;
    primary: string;
    primaryVariant: string;
    secondary: string;
    secondaryVariant: string;
    inactive: string;
    danger: string;
    error: string;
    warning: string;
  };
};

export const lightModeAppTheme: AppTheme = {
  name: 'lightTheme',
  colors: {
    background: '#ffffff',
    backgroundHeader: '#fafafa',
    backgroundBottomTab: '#fafafa',
    backgroundSnackbar: '#404040',
    text: '#212935',
    textInverted: '#ffffff',
    textStrong: '#000000',
    textWeak: '#9e9e9e',
    border: '#e0e0e0',
    borderStrong: '#9e9e9e',
    primary: '#3a6be8',
    primaryVariant: '#85a3f1',
    secondary: '#e8b73a',
    secondaryVariant: '#f8f3dd',
    inactive: '#e0e0e0',
    danger: '#d33246',
    error: '#d33246',
    warning: '#e8b73a',
  },
};

export const darkModeAppTheme: AppTheme = {
  ...lightModeAppTheme,
  name: 'darkTheme',
};
