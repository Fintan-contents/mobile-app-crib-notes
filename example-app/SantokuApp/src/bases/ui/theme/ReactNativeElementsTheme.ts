import {AppTheme} from './AppTheme';

export const getReactNativeElementsTheme = (appTheme: AppTheme) => {
  const colors = appTheme.colors;
  return {
    // Buttonはタイプによってスタイルが変わるのでここでは設定しない
    Text: {
      style: {
        color: colors.text,
      },
      h1Style: {
        color: colors.textStrong,
      },
      h2Style: {
        color: colors.textStrong,
      },
      h3Style: {
        color: colors.textStrong,
      },
      h4Style: {
        color: colors.textStrong,
      },
    },
    Input: {
      style: {
        fontSize: 17,
        color: colors.text,
        includeFontPadding: false,
      },
      labelStyle: {
        fontSize: 15,
        fontWeight: 'normal',
        color: colors.text,
      },
      containerStyle: {
        paddingHorizontal: 0,
      },
      inputContainerStyle: {
        borderBottomColor: colors.borderStrong,
        borderBottomWidth: 2,
      },
      errorStyle: {
        fontSize: 12,
        color: colors.error,
        includeFontPadding: false,
      },
    },
  };
};
