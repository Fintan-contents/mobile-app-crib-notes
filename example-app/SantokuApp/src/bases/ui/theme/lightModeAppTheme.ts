import {AppTheme} from './AppTheme';

export const lightModeAppTheme: AppTheme = {
  name: 'lightTheme',
  colors: {
    background: '#ffffff',
    backgroundHeader: '#fafafa',
    backgroundBottomTab: '#fafafa',
    backgroundSnackbar: '#404040',
    text: '#212935',
    textInverted: '#ffffff',
    textStrong: '#000000',
    textWeak: '#9e9e9e',
    border: '#e0e0e0',
    borderStrong: '#9e9e9e',
    primary: '#3a6be8',
    primaryVariant: '#85a3f1',
    secondary: '#e8b73a',
    secondaryVariant: '#f8f3dd',
    inactive: '#e0e0e0',
    danger: '#d33246',
    error: '#d33246',
    warning: '#e8b73a',
  },
};
