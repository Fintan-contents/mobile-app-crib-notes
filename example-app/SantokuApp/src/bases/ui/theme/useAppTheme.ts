import {createUseContextAndProvider} from 'bases/core/contexts/createUseContextAndProvider';

import {AppTheme} from './AppTheme';

export const [useAppTheme, AppThemeContextProvider] = createUseContextAndProvider<AppTheme>();
