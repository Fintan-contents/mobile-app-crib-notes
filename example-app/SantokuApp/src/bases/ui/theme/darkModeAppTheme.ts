import {AppTheme} from './AppTheme';
import {lightModeAppTheme} from './lightModeAppTheme';

export const darkModeAppTheme: AppTheme = {
  ...lightModeAppTheme,
  name: 'darkTheme',
};
