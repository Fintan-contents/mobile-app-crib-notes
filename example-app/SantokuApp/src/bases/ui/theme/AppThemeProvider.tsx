import {ThemeProvider as RestyleThemeProvider} from '@shopify/restyle';
import React, {useMemo} from 'react';
import {ColorSchemeName, useColorScheme} from 'react-native';
import {ThemeProvider as RNEThemeProvider} from 'react-native-elements';

import {getReactNativeElementsTheme} from './ReactNativeElementsTheme';
import {darkModeAppTheme} from './darkModeAppTheme';
import {lightModeAppTheme} from './lightModeAppTheme';
import {restyleTheme} from './restyleTheme';
import {AppThemeContextProvider} from './useAppTheme';

const getAppTheme = (colorScheme: ColorSchemeName) => (colorScheme === 'dark' ? darkModeAppTheme : lightModeAppTheme);

const AppThemeProvider: React.FC<React.PropsWithChildren> = ({children}) => {
  const colorScheme = useColorScheme();
  const appTheme = getAppTheme(colorScheme);
  const reactNativeElementsTheme = useMemo(() => getReactNativeElementsTheme(appTheme), [appTheme]);

  return (
    <AppThemeContextProvider value={appTheme}>
      <RestyleThemeProvider theme={restyleTheme}>
        <RNEThemeProvider theme={reactNativeElementsTheme}>{children}</RNEThemeProvider>
      </RestyleThemeProvider>
    </AppThemeContextProvider>
  );
};

export {AppThemeProvider};
