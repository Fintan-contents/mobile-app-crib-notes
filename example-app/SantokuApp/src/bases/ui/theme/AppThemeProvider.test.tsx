import {render, renderHook, screen} from '@testing-library/react-native';
import {AppTheme} from 'bases/ui/theme/AppTheme';
import {AppThemeContextProvider, useAppTheme} from 'bases/ui/theme/useAppTheme';
import React from 'react';
import {Text} from 'react-native';

import {AppThemeProvider} from './AppThemeProvider';
import {darkModeAppTheme} from './darkModeAppTheme';
import {lightModeAppTheme} from './lightModeAppTheme';

const wrapper = (value: AppTheme) => {
  return ({children}: {children: React.ReactNode}) => {
    return <AppThemeContextProvider value={value}>{children}</AppThemeContextProvider>;
  };
};

describe('AppThemeProvider', () => {
  const ChildComponent: React.FC = () => {
    return <Text testID="test">test</Text>;
  };

  it('AppThemeProviderを子要素を含めて正常にrenderできること', () => {
    render(
      <AppThemeProvider>
        <ChildComponent />
      </AppThemeProvider>,
    );
    expect(screen.queryByTestId('test')).not.toBeNull();
    expect(screen).toMatchSnapshot();
  });
});

describe('useAppTheme', () => {
  it('AppThemeProviderのテーマがlight用の場合にlight用のAppThemeが取得できること', () => {
    const {result} = renderHook(() => useAppTheme(), {wrapper: wrapper(lightModeAppTheme)});
    const appTheme = result.current;
    expect(appTheme).toEqual(lightModeAppTheme);
  });

  it('AppThemeProviderのテーマがdark用の場合にdark用のAppThemeが取得できること', () => {
    const {result} = renderHook(() => useAppTheme(), {wrapper: wrapper(darkModeAppTheme)});
    const appTheme = result.current;
    expect(appTheme).toEqual(darkModeAppTheme);
  });
});
