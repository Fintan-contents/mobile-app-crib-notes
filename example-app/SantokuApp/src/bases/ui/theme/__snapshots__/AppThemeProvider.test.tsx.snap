// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppThemeProvider AppThemeProviderを子要素を含めて正常にrenderできること 1`] = `
<Text
  testID="test"
>
  test
</Text>
`;
